website/content/en/docs/kubebuilder/quickstart.md

```diff
@@ -0,0 +1,763 @@
+  ...
+}
+```
+
+Based on the return values, [`Result`][result_go_doc] and error, the `Request` may be requeued and the reconcile loop may be triggered again:
+
+```Go
+// Reconcile successful - don't requeue
+return ctrl.Result{}, nil
+// Reconcile failed due to error - requeue
+return ctrl.Result{}, err
+// Requeue for any reason other than error
+return ctrl.Result{Requeue: true}, nil
+```
+
+You can set the `Result.RequeueAfter` to requeue the `Request` after a grace period as well:
+```Go
+import "time"
+
+// Reconcile for any reason than error after 5 seconds
+return ctrl.Result{RequeueAfter: time.Second*5}, nil
+```
+
+**Note:** Returning `Result` with `RequeueAfter` set is how you can periodically reconcile a CR.
+
+For a guide on Reconcilers, Clients, and interacting with resource Events, see the [Client API doc][doc_client_api].
+
+> **// TODO:** Should we point to the client API doc after the integration?
+
+### Specify permissions and generate RBAC manifests
+
+The controller needs certain RBAC permissions to interact with the resources it manages. These are specified via [RBAC markers][rbac_markers] like the following:
+
+```Go
+// +kubebuilder:rbac:groups=cache.example.com,resources=memcacheds,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=cache.example.com,resources=memcacheds/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;
+
+func (r *MemcachedReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+```
+
+The `ClusterRole` manifest at `config/rbac/role.yaml` is generated from the above markers via controller-gen with the following command:
+
+```sh
+$ make manifests
+```
+
+## Build and run the operator
+
+Before running the operator, the CRD must be registered with the Kubernetes apiserver:
+
+```sh
+$ make install
+```
+
+Once this is done, there are two ways to run the operator:
+
+- As Go program outside a cluster
+- As a Deployment inside a Kubernetes cluster
+
+### 1. Run locally outside the cluster
+
+To run the operator locally execute the following command:
+
+```sh
+$ make run ENABLE_WEBHOOKS=false
+```
+
+> **// TODO:** Add an advanced section or separate doc on writing defaulting/validating and conversion webhooks.
+
+### 2. Run as a Deployment inside the cluster
+
+#### Build and push the image
+
+Build the operator image and push it to a registry.
+
+Make sure to modify `quay.io/example/` in the example below to reference a container repository that
+you have access to. You can obtain an account for storing containers at
+repository sites such quay.io or hub.docker.com
+
+Build the image:
+
+```sh
+$ make docker-build IMG=quay.io/example/memcached-operator:v0.0.1
+```
+
+Push the image to a repository:
+
+```sh
+$ make docker-push IMG=quay.io/example/memcached-operator:v0.0.1
+```
```

Comment:
Yeah this is better. 
Makes it so that they can't blindly copy paste to build but then be unable to push the image.

I'll still add a line about replacing the image line completely depending on the image repository that they're using.